### DISCUSSION TITLE
Running srsLTE at ZC702 and ZC706

### DISCUSSION THREAD

**Discussion by iokarapa** [srsran/zynq_timestamping#48] [Ideas]
I need to integrate a 4G/5G system using srsLTE in a virtual machine connected to ZC702 and ZC706 boards with AD-FMCOMMS5-EBZ. I am having trouble with my setup and suspect the problem may be with the timestamping of the boards. If the timestamping solution can be ported or adapted to both ZC702 and ZC706 would be appreciated. 
Thank you for your work, and I hope for the best.

**Comment by ofontbach** [comment]
Hi @iokarapa,

If I get it right, the problem with your setup is that you can't find a pre-built version of the zynq timestamping solution for neither ZC702 nor ZC706, so with the default ADI firmware for those boards there is no timestamping implemented. I'm sure, porting the solution to those boards would be quite useful to the community.

You can base your porting on the implementation we are providing for the [ZCU102 platform](https://github.com/srsran/zynq_timestamping/tree/main/projects/zcu102). You could build the Vivado project as described in the [documentation](https://srsran.github.io/zynq_timestamping/bitstream/zcu102_project.html) and inspect the block design to understand how to modify the base firmware that ADI provides for the AD-FMCOMMS5-EBZ. For instance, you could modify the [TCL script to build the project for the ZCU102](https://github.com/srsran/zynq_timestamping/blob/main/projects/zcu102/src/bd/design_1.tcl) for your goals.

A few useful pointers:
- Base ADI firmware for [AD-FMCOMMS2 + ZCU102](https://github.com/analogdevicesinc/hdl/tree/43cdc6263baf3edb166a3def6fab15bc81c4c729/projects/fmcomms2/zcu102) and [AD-FMCOMMS5 + ZCU102](https://github.com/analogdevicesinc/hdl/tree/43cdc6263baf3edb166a3def6fab15bc81c4c729/projects/fmcomms5/zcu102). We use this 
- Base ADI firmware for [AD-FMCOMMS2 + ZC702](https://github.com/analogdevicesinc/hdl/tree/43cdc6263baf3edb166a3def6fab15bc81c4c729/projects/fmcomms2/zc702) and [AD-FMCOMMS5 + ZC702](https://github.com/analogdevicesinc/hdl/tree/43cdc6263baf3edb166a3def6fab15bc81c4c729/projects/fmcomms5/zc702).
- Base ADI firmware for [AD-FMCOMMS2 + ZC706](https://github.com/analogdevicesinc/hdl/tree/43cdc6263baf3edb166a3def6fab15bc81c4c729/projects/fmcomms2/zc706) and [AD-FMCOMMS5 + ZC706](https://github.com/analogdevicesinc/hdl/tree/43cdc6263baf3edb166a3def6fab15bc81c4c729/projects/fmcomms5/zc706).

Once a new bitstream is ready, you'd also need to adapt the device tree and prepare a new BOOT.BIN which you can also do by following our [documentation](https://srsran.github.io/zynq_timestamping/app/zcu.html) (Petalinux way) or [ADI's documentation](https://wiki.analog.com/resources/eval/user-guides/ad-fmcomms2-ebz/quickstart/zynqmp).

Let us know how it goes for your or ask more specific questions if you find problems during the porting.

Thanks!

**Reply by Salex406** [reply]
Hello. 
Could you please explain, your ZCU102 project only supports FMCOMMS5 analog board, am I right? So, if I want to use with another board  (FMCDAQ2), I need to take hdl from ADI and customize?

**Reply by ofontbach** [reply]
Hi @Salex406,

The [ZCU102 project in our repository](https://github.com/srsran/zynq_timestamping/tree/main/projects/zcu102) is meant to be used with the FMCOMMS2/3/4. Above I'm pointing both to the baseline ADI HDL project (i.e., without timestamping) for that setup, as well as to its counterpart for the FMCOMMS5, while also referring to the equivalent projects for the ZC702/6.

If you want to port our solution to add suppor to another RF front-end, then yes, you'd need to add our solution on top of the original HDL project provided by ADI. Our recommendation is to generate the Vivado project we are providing for a similar setup (e.g., zcu102 + fmcomms2/3/4) and inspect it in order to apply the required modifications to ADI's HDL project (e.g., look at the block configuration and its interconnection with our IPs, to then either create a new TCL script taking ours as a basis, or create a new project using Vivado itself). As you will see, our repo and our scripts use ADI's HDL as a submodule and combine their HDL with ours; the same applies on the software end.

**Comment by angrammenos97** [comment]
Hello Community,
@iokarapa did you eventually manage to port the implementation to the zc702/6 FPGA?
I am also trying to port the solution on those FPGAs using FMCOMMS5 SDR. After importing the srs IPs into the original ADI's HDL project, having as references both zcu102 and pluto solutions, I tried the txrx example and I get the following error:
`zynq_timestamping/sw/lib/src/phy/rf/rf_iio_imp.c:796: Error refilling buf -9`
Do you have any clue what am I missing?
Thank you for your time.
